### PR TITLE
UX: Chat unread indicator refactor

### PR DESCRIPTION
**Pull request by chapoi**
## desktop
image
image
image
image

## mobile
image
image
image

## changes
* move the chat unread indicator to top to match the profile avatar indicator
* add white border to profile avatar indicator (badge notification) to match chat indicator and userstatus styling
* change `.urgent` to BEM
* congregate all styling into mixin
  * update chat index to use mixin
  * update thread indicator to use mixin
  * update header indicator to use mixin

### CODE REVIEW

File: plugins/chat/assets/javascripts/discourse/components/chat-channel-unread-indicator.hbs [hunk 1]
```diff
@@ -6,10 +6,12 @@
         (or
           @channel.isDirectMessageChannel (gt @channel.tracking.mentionCount 0)
         )
-        "urgent"
```
Comment (jjaffeux) [inline plugins/chat/assets/javascripts/discourse/components/chat-channel-unread-indicator.hbs:9]: it's always a little bit unclear to me when to use "-urgent" or "is-urgent"

Comment (chapoi) [reply plugins/chat/assets/javascripts/discourse/components/chat-channel-unread-indicator.hbs:9]: I know, this case is very ambiguous. I use is- and has- for temporary/toggle states, while urgent isn't a temporary addition, it's a different iteration of the regular notification. (All notifications are temporary of course, but that's besides the context here).
Does that make sense?

Comment (jjaffeux) [reply plugins/chat/assets/javascripts/discourse/components/chat-channel-unread-indicator.hbs:9]: Yes ok i see. So I would say it like this.

- `is-foo`/`has-foo` is a state of the component
- `-foo` is a variation of the component

Comment (chapoi) [reply plugins/chat/assets/javascripts/discourse/components/chat-channel-unread-indicator.hbs:9]: As I said, it's fairly ambiguous here. `has-urgent` just sounds wrong xD Would be `is-urgent` than I suppose. I'm not opposed but to me, we named it urgent, but we might as well could have named it "green" or "personal-dms-and-mentions". Which would be more like a variation again.

Comment (jjaffeux) [reply plugins/chat/assets/javascripts/discourse/components/chat-channel-unread-indicator.hbs:9]: Im almost thinking that state should in a data attribute. But I hear you in this case that `urgent` might be legit a variation of the component ( and `-urgent`) now that I think more about it.

**Review by martin-brennan — APPROVED** [review]

**Pull request merged**